code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -45,6 +46,17 @@
 /mob/living/simple_animal/hostile/cockroach/add_cell_sample()
 	AddElement(/datum/element/swabable, CELL_LINE_TABLE_COCKROACH, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 7)
 
+/mob/living/simple_animal/hostile/cockroach/Life(delta_time = SSMOBS_DT, times_fired) //Natural predators
+	. = ..()
+	turns_since_scan++
+	if(turns_since_scan > 5)
+		turns_since_scan = 0
+		var/list/target_types = list(/obj/effect/decal/cleanable/ants)
+		for(var/obj/effect/decal/cleanable/potential_target in view(4, get_turf(src)))
```

Comment:
have you profiled this at all? cockroaches are somewhat common and view(4,...) isnt trivial to spam

Reply:
How do I profile it? I'll admit I'm not the most knowledgeable when it comes to coding in here.

Reply:
you can use the inbuilt byond profiler by clicking on the tgstation icon in the top left of the ss13 window and selecting it. just sort by say Self CPU and see if you can find cockroach/Life() anywhere near the top (i forget if any get spawned roundstart, if not then spawn ~50)

Reply:
I ran it for around 7-10 minutes or so after spawning in 50 roaches like you said, I will admit it's pretty high up there.
![image](https://user-images.githubusercontent.com/66052067/124828408-392e3880-df45-11eb-9c82-33e7ce6941ca.png)
The pic is pretty small but all you gotta really see is the highlighted part in relation to where the scrollbar is. It's pretty high up.
I'd love to lower its usage but I'll admit I don't really know how I'd make them scan a radius around them in a more efficient manner than this. If there's a way for them to _only_ look for their target mob in that radius and ignore everything else, I'd love to hear it.

Reply:
i would lower the view radius of both by at least 1, while its not as bad as i was fearing at first there really isnt much of a reason for them to look through that big of a radius for something that doesnt interact with players.

Reply:
That's good to hear, I'll be sure to lower their ranges from 3 to 2.